satpy/writers/cf_writer.py

```diff
@@ -251,32 +180,25 @@ def area2lonlat(dataarray):
                         name='latitude')
     dataarray['longitude'] = lons
     dataarray['latitude'] = lats
-    return [dataarray]
 
 
 def area2gridmapping(dataarray):
     """Convert an area to at CF grid mapping."""
     area = dataarray.attrs['area']
-    attrs = create_grid_mapping(area)
-    if attrs is not None and 'name' in attrs.keys() and attrs['name'] != "proj4":
-        dataarray.attrs['grid_mapping'] = attrs['name']
-        name = attrs['name']
-    else:
-        # Handle the case when the projection cannot be converted to a standard CF representation or this has not
-        # been implemented yet.
-        dataarray.attrs['grid_proj4'] = area.proj4_string
-        name = "proj4"
-    return [dataarray, xr.DataArray(0, attrs=attrs, name=name)]
+    gmapping_var_name, attrs = create_grid_mapping(area)
+    dataarray.attrs['grid_mapping'] = gmapping_var_name
+    return xr.DataArray(0, attrs=attrs, name=gmapping_var_name)
 
 
 def area2cf(dataarray, strict=False):
     """Convert an area to at CF grid mapping or lon and lats."""
     res = []
     dataarray = dataarray.copy(deep=True)
     if isinstance(dataarray.attrs['area'], SwathDefinition) or strict:
-        res = area2lonlat(dataarray)
+        # modifies dataarray in-place
+        area2lonlat(dataarray)
```

Comment:
For style, I'd rather avoid in-place modifications. Can we make this return a modified copy of the array and do:
```suggestion
        dataarray = area2lonlat(dataarray)
```
?

Reply:
I know we had this before, but since we're changing this anyway...

Reply:
Sure. The way this is set up is that `area2lonlat` and `area2gridmapping` are given a copy by `area2cf`. So these two functions are in-place but are only called with a copy. Still want the change? Not a big deal.

Reply:
yes, I'd rather have each function make a copy.

Reply:
In the lonlat case we add coordinates, does that still only need a shallow copy?

Reply:
I think so, but don't take my word for it.